source/renderer/app/components/wallet/tokens/wallet-token/WalletToken.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React, { useState } from 'react';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import styles from './WalletToken.scss';
+import AssetContent from '../../../assets/AssetContent';
+import type { AssetToken } from '../../../../api/assets/types';
+import WalletTokenFooter from './WalletTokenFooter';
+import WalletTokenHeader from './WalletTokenHeader';
+
+type Props = {
+  anyAssetWasHovered: boolean,
+  asset: AssetToken,
+  assetSettingsDialogWasOpened: boolean,
+  className?: string,
+  headerClassName?: string,
+  fullFingerprint?: boolean,
+  isFavorite: boolean,
+  isInsertingAsset: boolean,
+  isLoading: boolean,
+  isRemovingAsset: boolean,
+  onAssetSettings?: Function,
+  onCopyAssetParam?: Function,
+  onOpenAssetSend?: Function,
+  onToggleFavorite?: Function,
+};
+
+const WalletToken = observer((props: Props) => {
+  const {
+    anyAssetWasHovered,
+    asset,
+    assetSettingsDialogWasOpened,
+    className,
+    headerClassName,
+    fullFingerprint = true,
+    isFavorite,
+    isLoading,
+    onAssetSettings,
+    onCopyAssetParam,
+    onOpenAssetSend,
+    onToggleFavorite,
+    isInsertingAsset,
+    isRemovingAsset,
+  } = props;
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const toggleIsExpanded = () => {
+    setIsExpanded(!isExpanded);
+  };
+  const componentStyles = classNames(
```

Comment:
useMemo here to prevent unnecessary execution when asset is hovered

Reply:
Right, I'll update. Thanks!